paddle/fluid/operators/batch_norm_mkldnn_op.cc

```diff
@@ -113,9 +113,16 @@ class BatchNormMKLDNNOpKernel : public paddle::framework::OpKernel<T> {
     if (is_test) flags |= mkldnn::use_global_stats;
 
     // create mkldnn memory from input x tensor
-    auto src_memory =
-        memory({{{src_tz}, memory::data_type::f32, x->format()}, mkldnn_engine},
-               to_void_cast(x_data));
+    mkldnn::memory::format input_format = x->format();
+    if (src_tz.size() == 1) {
+      input_format = mkldnn::memory::format::x;
+    } else if (src_tz.size() == 2) {
+      input_format = mkldnn::memory::format::nc;
+    }
```

Comment:
You can use a function to package following codes, since they are duplicated three times in this file.
```
if (src_tz.size() == 1) {
  input_format = mkldnn::memory::format::x;
    } else if (src_tz.size() == 2) {
      input_format = mkldnn::memory::format::nc;
    }
```

Reply:
Sure, I have been changing it.